resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/Config.scala

```diff
@@ -0,0 +1,115 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.deploy.k8s
+
+import org.apache.spark.SPARK_VERSION
+import org.apache.spark.internal.Logging
+import org.apache.spark.internal.config.ConfigBuilder
+import org.apache.spark.network.util.ByteUnit
+
+private[spark] object Config extends Logging {
+
+  val KUBERNETES_NAMESPACE =
+    ConfigBuilder("spark.kubernetes.namespace")
+      .doc("The namespace that will be used for running the driver and executor pods. When using" +
+        " spark-submit in cluster mode, this can also be passed to spark-submit via the" +
+        " --kubernetes-namespace command line argument.")
+      .stringConf
+      .createWithDefault("default")
+
+  val EXECUTOR_DOCKER_IMAGE =
+    ConfigBuilder("spark.kubernetes.executor.docker.image")
+      .doc("Docker image to use for the executors. Specify this using the standard Docker tag" +
+        " format.")
+      .stringConf
+      .createWithDefault(s"spark-executor:$SPARK_VERSION")
```

Comment:
I think we should remove this instead of defaulting. WDYT @mccheah

Reply:
I agree.  Removed in https://github.com/apache/spark/pull/19468/commits/4bed817e5ab1a26e70050254fec453e839af4c8c.

Reply:
I'm somewhat concerned about this though.
IMO one of key value we delivery with this feature is running on k8s out of the box, and the Docker images we are going released with Spark is a key aspect of that (again, as discussed, we should highlight that this SPIP/feature is going to deliver and release that)

Secondly, it is going to be very easy to make a mistake and set the "wrong" Docker images (there are many many ways this could go wrong, including mismatch versions of submission client vs driver &/ executor) that would be very hard to diagnose (by no means it will fail fast - worse case you get some subtle correctness issues in data output). And IMO having this version number set consistently by default is really going to help that.

What's the concern with having the default value?

Reply:
The concern is the availability of this default image. If we can make a strong guarantee by making tagging and publication of the images part of the release process, this will not be an issue. @mridulm @foxish @mccheah thoughts?

Reply:
IMO we should, as a part of releasing Spark with this feature, to have Docker image in the same release. Granted, it is not a guarantee, but given what we know about building Docker images and Docker image publishing supported officially by the ASF / ASF Infra, this shouldn't be a problem.

On the flip side if we couldn't have "official" or default images, then we should consider how to build some validations in the startup path such that "wrong" or mismatch images can be detected quickly / fail-fast.

Reply:
@felixcheung That would be a behavior change w.r.t how we have used docker till now in spark - we expect users to explicitly specify the image.
We should not be introducing default behavior for a new feature without adequately understanding the implications of it - we can always introduce a smart default in a future spark version as required with minimal impact to end users.

I do agree that it is a good idea to validate version, etc - irrespective of whether we have a default value or not. It will help not just k8s integration, but also mesos and spark on yarn + docker.

For now, we can fail fast when value is not specified.